libgnucash/backend/xml/gnc-bill-term-xml-v2.cpp

```diff
@@ -151,12 +151,14 @@ static gboolean
 set_numeric (xmlNodePtr node, GncBillTerm* term,
              void (*func) (GncBillTerm*, gnc_numeric))
 {
-    gnc_numeric* num = dom_tree_to_gnc_numeric (node);
-    g_return_val_if_fail (num, FALSE);
+    gnc_numeric num;
+    gboolean ret;
```

Comment:
This is C++: *Always* declare and initialize variables in the same line, and unless there's a scoping reason do so at first use. If you need a variable to be set in a scope and used outside of it declare and init just before the beginning of the scope where it will be set.

Also C++ has a `bool` type. Don't use `gboolean` unless interfacing with GLib or Gtk.

Do the same in new C code. We've removed the C89 restriction on mixing decls and code, and C99 has a `bool` type (you have to `#include <bool.h>` to use it).

Reply:
As for changing `dom_tree_to_gnc_numeric` from returning a `gnc_numeric*` to returning a `bool` with an in/out reference parameter, *eww*. Just return a `gnc_numeric`.  [Copy elision](https://en.wikipedia.org/wiki/Copy_elision) is mandatory in C++17 so the numeric will get created in place in the calling function's stack.

Reply:
What do you want it to do when `dom_tree_to_text` in `dom_tree_to_gnc_numeric` returns NULL?

Reply:
The general modern C++ way is `std::optional` but that's not necessary here. You could return `GNC_ERROR_ARG`, but consider that `dom_tree_to_gnc_numeric` returns `gnc_numeric_zero` if `dom_tree_to_text` returns an empty string instead of nullptr. nullptr and "" are equivalent in this context so you might as well return 0 in both cases.

Reply:
Done